### ISSUE TITLE
Getters for "d *Document" attribute.

### ISSUE THREAD

**Issue by freb** [baliance/gooxml#259]
## Description

All structs have a getter for `x`, to get the raw xml, but none have a getter for `d`, the document. I'm in the middle of writing a bunch of helper functions, and for any that require looping through the document elements (which is most of them), you have to pass the document in as well, even though it is already included in whatever parameter you are accepting (Paragraph, Cell, Table, etc.). 

## Expected Behavior

Permit calling `.D()` on structs to get the underlying `*Document`.

## Actual Behavior

A getter does not exist.

**Comment by gunnsth** [comment]
Can you provide an example? Seems to me that the Document would usually be accessible pretty easily? Even if need to provide as an extra parameter.  Could even embed document in a new type that the helpers methods are defined on.  Would be good to have an example to support the case.

**Comment by freb** [comment]
There are no examples that cannot be worked around with the methods you suggest.

I reviewed the functions where I required a reference to `*Document` and discovered that they all relate to finding content within an existing document. I realized that I mainly have a need for these functions because there is no way to go from one element in the document to surrounding elements. For example, given a Cell, there is no way to get back to the row or table without looping through the  entire document (or maintaining a reference to those if you arrived at the cell from within a loop). Or to go from a paragraph to the next content block (paragraph or table), etc.

Having to do everything from within the same master loop in order to have access to adjacent document elements makes things a bit messy. Similarly, having to re-loop the entire document looking for adjacent elements is really inefficient.

I realize this explanation is a separate issue but I think it gets at the reason for me requesting `.D()` in the first place, even though access to the document in the way I requested is not all that useful in solving these larger issues.

I can close this issue, but I'd be interested in your thoughts about adding `InsertParagraphAfter` (similar to InsertRunAfter), `GetNextParagraph`, etc.
